internals/overlord/logstate/clienterr/clienterr.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+/*
+Package clienterr contains common error types which are recognised by the log
+gatherer. logstate.logClient implementations should return these error types
+to communicate with the gatherer.
+*/
+package clienterr
```

Comment:
Lacking a lot of context here but is it worth splitting the package into `client/error` instead of `clienterr`?

Just thinking aloud because in my head `clienterr` feels like an alias the consumer applies.

Reply:
I would say no, there's no need for a separate `client` package at this stage